Zend/zend_type_info.h

```diff
@@ -64,6 +64,11 @@
 #define MAY_BE_ARRAY_KEY_STRING     (1<<23)
 #define MAY_BE_ARRAY_KEY_ANY        (MAY_BE_ARRAY_KEY_LONG | MAY_BE_ARRAY_KEY_STRING)
 
+#define MAY_BE_PACKED(t)            ((t) & MAY_BE_ARRAY_PACKED)
+#define MAY_BE_HASH(t)              ((t) & (MAY_BE_ARRAY_HASH | MAY_BE_ARRAY_KEY_STRING))
+#define MAY_BE_PACKED_ONLY(t)       (MAY_BE_PACKED(t) && !MAY_BE_HASH(t))
+#define MAY_BE_HASH_ONLY(t)         (MAY_BE_HASH(t) && !MAY_BE_PACKED(t))
```

Comment:
Can we rename `MAY_BE_ARRAY_PACKED`/`MAY_BE_ARRAY_HASH` to `MAY_BE_ARRAY_LONG_PACKED`/`MAY_BE_ARRAY_LONG_HASH`? I was somewhat confused here until I understood that PACKED/HASH only refer to integer keys and not string keys.

Reply:
> Can we rename `MAY_BE_ARRAY_PACKED`/`MAY_BE_ARRAY_HASH` to `MAY_BE_ARRAY_LONG_PACKED`/`MAY_BE_ARRAY_LONG_HASH`? I was somewhat confused here until I understood that PACKED/HASH only refer to integer keys and not string keys.

Agree. During implementation I was confused myself. :)